### PR TITLE
[ADP-3407] Bump to ghc 9.8.2

### PR DESCRIPTION
**Pull request by paolino**
- [x] Bump int-cast
- [x] Bump openapi3
- [x] Remove all constraints in cabal files
- [x] Bump compiler in haskell.nix to 9.8.2

### CODE REVIEW

File: cabal.project [hunk 1]
```diff
@@ -157,6 +157,13 @@ source-repository-package
     subdir: generated
 
 --------------------------------------------------------------------------------
+
+source-repository-package
+    type: git
+    location: https://github.com/paolino/int-cast.git
+    tag: 0232b037552a4247b4a7ac21f816cc77cc95f7a4
+    --sha256: 1pmghd9pz35ycxp0c3f5sz4fghfi6ywjmp6wm9n45f8l5ah9vhhw
```
Comment (jonathanknowles) [inline cabal.project:185]: I've just made a PR to upgrade `int-cast` to support GHC versions 9.8 and 9.10 here:
- https://github.com/haskell-hvr/int-cast/pull/9

File: cabal.project [hunk 2]
```diff
@@ -177,8 +184,7 @@ allow-newer:
 
 
 constraints:
-    base == 4.18.2.0
-  , bimap >= 0.4.0
+  bimap >= 0.4.0
```
Comment (jonathanknowles) [inline cabal.project:195]:
```suggestion
  , bimap >= 0.4.0
```

**Review by jonathanknowles — APPROVED** [review]
LGTM!

I've tested it locally, and:
- everything seems to build just fine. 🎉
- HLS continues to work nicely. 🎉 

Hopefully it will also work for others too!

**Review by Anviking — APPROVED** [review]
`cabal build all` in nix shell and HLS works on my mac

**Review by HeinrichApfelmus — APPROVED** [review]
> cabal build all in nix shell and HLS works on my mac

Same for me. 👍

**Review by paweljakubas — APPROVED** [review]
:+1:

**Comment by abailly** [comment]
We should probably bump to whatever version the currently supported node (10.1.3) is using, which is not totally obvious from their repository...

**Comment by abailly** [comment]
current supported ghc version for 10.1.3 is 8.10.7: https://github.com/IntersectMBO/cardano-node/releases/tag/10.1.3
Closing this outdated RP

**Pull request closed without merging**